src/main/java/Terminator.java

```diff
@@ -25,14 +41,58 @@
      */
     public static ArrayList<Task> tasksList = new ArrayList<Task>();
 
+    /**
+     * Given user input, extract Task Name and UserInput.
+     * @param userinput The string given by the user.
+     * @param eventType The type of task to extract based on.
+     * @return String array containing userinput and eventType.
+     */
+    public static String[] extractNameDateTime(String userinput, String eventType){
+        String[] returnArray = new String[2];
+        if (Objects.equals(eventType, DEADLINE_TYPE)){
+            // Get indexes to substring
+            int startOfByIndex = userinput.indexOf("/by");
+            int endOfByIndex = startOfByIndex + 4;
+            int endOfDeadlineStringIndex = userinput.indexOf("deadline") + 9;
+
+            // Get specific task_name and date_time
+            String task_name = userinput.substring(endOfDeadlineStringIndex, startOfByIndex).strip();
+            String date_time = userinput.substring(endOfByIndex).strip();
+
+            // Place values into array
+            returnArray[TASK_NAME_INDEX] = task_name;
+            returnArray[DATE_TIME_INDEX] = date_time;
+        } else if (Objects.equals(eventType, EVENT_TYPE)){
+            // Get indexes to substring
+            int startOfAtIndex = userinput.indexOf("/at");
+            int endOfAtIndex = startOfAtIndex + 4;
+            int endOfEventStringIndex = userinput.indexOf("event") + 6;
+
+            // Get specific task_name and date_time
+            String task_name = userinput.substring(endOfEventStringIndex, startOfAtIndex).strip();
+            String date_time = userinput.substring(endOfAtIndex).strip();
```

Comment:
Variables name must be written in camelCase.
![image](https://user-images.githubusercontent.com/59493341/131812188-b56756dc-bb66-43d2-aeff-edfa1ff7c8f7.png)